File: Backend/API/src/main/java/com/example/localhistory/homework/dto/response/HomeworkDTO.java
package com.example.localhistory.homework.dto.response;

import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

// Response sent for homework definition reads and teacher management screens.
@Data
public class HomeworkDTO {
    private Long id;
    private String title;
    private String description;
    private LocalDate dueDate;
    private LocalDateTime dateCreated;
    private Boolean requireVisit;
    private Boolean requireQuiz;
    private Long landmarkId;
    private String landmarkTitle;
    private Long teacherId;
    private Integer assignmentsCount;
}

File: Frontend/app/src/main/java/com/example/localhistory/model/response/HomeworkDTO.kt
package com.example.localhistory.model.response

import kotlinx.datetime.LocalDate
import kotlinx.datetime.LocalDateTime

data class HomeworkDTO(
    val id: Long,
    val title: String,
    val description: String,
    val dueDate: LocalDate,
    val dateCreated: LocalDateTime?,
    val requireVisit: Boolean,
    val requireQuiz: Boolean,
    val landmarkId: Long,
    val landmarkTitle: String?,
    val teacherId: Long,
    val assignmentsCount: Int
)
